test/controllers/PropertyLinkingBaseControllerSpec.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import basespecs.BaseControllerSpec
+import play.api.http.HttpEntity.Strict
+import play.api.libs.json.{JsValue, Json, OFormat}
+import play.api.mvc.Result
+import play.api.test.FakeRequest
+
+import scala.concurrent.Future
+
+class PropertyLinkingBaseControllerSpec extends BaseControllerSpec {
+
+  trait Setup {
+
+    object Controller extends PropertyLinkingBaseController
+
+    case class Foo(bar: String)
+
+    implicit val format: OFormat[Foo] = Json.format
+
+    val function: Foo => Future[Result] =
+      _ => Future.successful(Ok)
+
+    val foo: Foo = Foo(bar = "FooName")
+    val fooJson: JsValue = Json.toJson(foo)
+    val barJson: JsValue = Json.obj()
+  }
+
+  "withJsonBody" should {
+    "evaluate function body" when {
```

Comment:
is this not testing logic from a library/

Reply:
no, we don't use the library stuff, we have our own withJsonBody in the base controller, and wasn't covered by tests